urlc58b5c5b73_emiliano_fiuba_2013_2c_7510_tgz-pJ8-test_java_FrameworkTestsJ8/src/main/java/ResultPrinter.java

```diff
@@ -0,0 +1,82 @@
+package main.java;
+/**
+ * This class works as the user visual interface of the testing framework, 
+ * printing the report of the tests executed and their results
+ * 
+ * @author Miguenz, Rodriguez, Suárez
+ *
+ */
+
+import java.util.ArrayList;
+
+public class ResultPrinter {
+
+	private String SEPARATOR = "----------------";
+	private String DOUBLE_SEPARATOR = "================";
+	private String SUMMARY_HDR = "[failure] Summary";
+	private static ResultPrinter instance = null;
+	private String suiteName;
+	private ArrayList<String> buffer;
+	private int failedTests;
+	private int errorTests;
+	private int okTests;
+	
+	private ResultPrinter() {
+		suiteName = "";
+		failedTests = 0;
+		errorTests = 0;
+		okTests = 0;
+		buffer = new ArrayList<String>();
+	}
+	
+	public static ResultPrinter getInstance() {
+		if (instance == null){
+			instance = new ResultPrinter();
+		}
+		return instance;
+	}
+
+	public void addTestResults(Test test) {
+		try {
+			buffer.add(test.getResult().print());
+			test.getResult().updateMe(this);
+		} catch (NullPointerException e) { }
+	}
+	
+	public void addSuite(String suiteName) {
+		this.suiteName += "." + suiteName;
+		buffer.add(this.suiteName);
+		buffer.add(SEPARATOR);
+	}
+	
+	public void removeSuite(String suiteName) {
+		//Remove the last added Suite name
+		this.suiteName.substring(0, this.suiteName.lastIndexOf("."));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 54 does not find the substring and returns -1, the call to substring() on line 54 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string suiteName.